lib/engine/step/share_buying.rb

```diff
@@ -17,12 +17,13 @@ def buy_shares(entity, shares, exchange: nil)
       # This works irrespective of if that player has sold this round
       # such as in 1889 for exchanging Dougo
       #
-      def can_gain?(entity, bundle)
+      def can_gain?(entity, bundle, exchanging = false)
```

Comment:
I wonder if 
exchange: false 
would be better?

Then it would be called with 
  exchange: true
which is a bit more expressive than just passing a simple true.